client/src/modules/lead/AddLead/AddLead.js

```diff
@@ -212,21 +213,26 @@ class AddLead extends React.Component {
     this.setState({
       ...this.state,
       name: this.getNameValue(),
+      organization: {
+        id: this.state.organizationAfterSelect.id,
+        name: trim(this.state.organization.name),
+      },
       namePlaceholder: this.getPlaceholderValue(),
       nameChanged: isBlank(this.state.organization.name) && isBlank(this.state.contact.name) ? false : this.state.nameChanged,
       openOrganizationDropdown: false,
-      showOrganizationBadge: this.state.organization.name.length > 0 && this.state.afterOrganizationSelectShowBadge && !this.state.organization.id,
+      showOrganizationBadge: this.state.organization.name.length > 0 && this.state.afterOrganizationSelectShowBadge && !this.state.organization.id && !isBlank(this.state.organization.name),
```

Comment:
2 same condition this.state.contact.name > 0 && !isBlank(this.state.contact.name)